src/main/java/de/retest/recheck/cli/subcommands/Diff.java

```diff
@@ -1,73 +1,169 @@
 	private List<String> exclude;
 
-	@Parameters( arity = "1", description = TestReportUtil.TEST_REPORT_PARAMETER_DESCRIPTION )
-	private File testReport;
+	@Option( names = "--help", usageHelp = true, hidden = true )
+	private boolean displayHelp;
 
-	@Override
-	public void run() {
-		if ( !PreCondition.isSatisfied() ) {
+	@Option( names = "--output", arity = "1",
+			description = "Save differences of Golden Masters to specified directory as test report." )
+	private Path directory;
+
+	@Override public void run() {
+		// until the following version, a user may still use the diff command to print test report differences.
+		// remove in next version
+		if ( goldenMasterPath.length == 1 ) {
+			logger.warn(
+					"The diff command for viewing a test report is deprecated and will be removed in the following version."
+							+ " Please use the new command 'show' to display a test report." );
+			final String[] args = { goldenMasterPath[0].toString() };
+			final Show c = new Show();
+			new CommandLine( c ).parseArgs( args );
+			c.run();
 			return;
 		}
+
+		final Path expectedSutStatePath = goldenMasterPath[0];
+		final Path actualSutStatePath = goldenMasterPath[1];
+
 		try {
-			final List<String> invalidFilters = FilterUtil.getInvalidFilters( exclude );
-			if ( !invalidFilters.isEmpty() ) {
-				FilterUtil.logWarningForInvalidFilters( invalidFilters );
-			} else {
-				printDiff();
+			if ( !hasValidExcludeOption() ) {
+				return;
 			}
+			final ActionReplayResult actionReplayResult =
+					createActionReplayResultFrom( expectedSutStatePath, actualSutStatePath );
+			final ActionReplayResult filteredResult = createFilteredActionReplayResult( actionReplayResult );
+			printGoldenMasterDifferences( filteredResult );
+			persistIfOutputOptionPresent( actionReplayResult );
 		} catch ( final Exception e ) {
 			ErrorHandler.handle( e );
 		}
 	}
 
-	private void printDiff() throws TestReportFormatException, IOException {
-		final TestReport report = TestReportUtil.load( testReport );
+	private boolean hasValidExcludeOption() {
+		final List<String> invalidFilters = FilterUtil.getInvalidFilters( exclude );
+		if ( !invalidFilters.isEmpty() ) {
+			FilterUtil.logWarningForInvalidFilters( invalidFilters );
+			return false;
+		}
+		return true;
+	}
+
+	private ActionReplayResult createActionReplayResultFrom( final Path expectedSutStatePath,
+			final Path actualSutStatePath ) {
+		final SutState expectedSutState = goldenMasterProvider
+				.loadGoldenMaster( createFile( expectedSutStatePath, RecheckProperties.DEFAULT_XML_FILE_NAME ) );
+		final SutState actualSutState = goldenMasterProvider
+				.loadGoldenMaster( createFile( actualSutStatePath, RecheckProperties.DEFAULT_XML_FILE_NAME ) );
+
+		return getActionReplayResult( expectedSutStatePath.toString(), expectedSutState, actualSutState );
+	}
+
+	private ActionReplayResult createFilteredActionReplayResult( final ActionReplayResult actionReplayResult ) {
```

Comment:
This is a bit shorter...
```suggestion
	private ActionReplayResult filterActionReplayResult( final ActionReplayResult actionReplayResult ) {
```

Reply:
Great suggestions, thanks!